src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -91,7 +93,21 @@ public IEnumerable<TagHelperDescriptor> GetDescriptors(string tagName, IEnumerab
                 {
                     foreach (var requiredAttribute in descriptor.RequiredAttributes)
                     {
-                        if (!attributeNames.Contains(requiredAttribute, StringComparer.OrdinalIgnoreCase))
+                        // '*' at the end of a required attribute indicates: apply to attributes prefixed with the
+                        // required attribute value.
+                        if (requiredAttribute.EndsWith(
+                            RequiredAttributeWildcardSuffix,
+                            StringComparison.OrdinalIgnoreCase))
+                        {
+                            var prefix = requiredAttribute.Substring(0, requiredAttribute.Length - 1);
+
+                            if (!attributeNames.Any(
+                                attributeName => attributeName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Should not successfully match the required attribute when `attributeName` is exactly equal to `prefix`. Such a match and a dictionary property with the same prefix will only lead to parse errors.